### PR TITLE
test: add Solana delta ingress witnessing test to bouncer

### PR DESCRIPTION
**Pull request by albert-llimos**
## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have written sufficient tests.
- [x] I have written and tested required migrations.
- [x] I have updated documentation where appropriate.

## Summary

Added test for Solana delta ingress witnessing. The test is failing at the moment, as expected:
https://github.com/chainflip-io/chainflip-backend/actions/runs/10401446608/job/28804557637
When this is fixed with the new pallet in PR #5060 we can add it as part of the bouncer tests.

### CODE REVIEW

File: bouncer/tests/delta_base_ingress.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      }
+      return false;
+    },
+    abortable: true,
+  });
+
+  // Initiate swap from Solana
+  const swapParams = await testSwap(
+    sourceAsset,
+    'Eth',
```
Comment (kylezs) [inline bouncer/tests/delta_base_ingress.ts:73]: ArbEth? just so it's faster?

File: bouncer/tests/delta_base_ingress.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  // Wait to ensure no new swap is being triggered after restart.
+  console.log('Waiting for 30 seconds to ensure no swap is being triggered after restart');
+  await sleep(30000);
+  swapScheduledHandle.stop();
+
+  if (swapsWitnessed !== 1) {
+    throw new Error('No swap was initiated. Swaps witnessed: ' + swapsWitnessed);
+  }
+
+  // Kill the engine
```
Comment (kylezs) [inline bouncer/tests/delta_base_ingress.ts:95]: think it's pretty clear what the function is doing 😆

File: bouncer/tests/delta_base_ingress.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  });
+  await startEngines(localnetInitPath, binariesPath, numberOfNodes);
+
+  // Wait to ensure no additional new swap is being triggered after restart
+  // and check that swap completes.
+  await sleep(30000);
+  await swapHandle;
+  swapScheduledHandle.stop();
+
+  if (swapsWitnessed < 2) {
```
Comment (kylezs) [inline bouncer/tests/delta_base_ingress.ts:137]: Having this condition twice, here and inside SwapScheduled is redundant (and confusing imo).

File: bouncer/tests/delta_base_ingress.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        inputAmount > Number(amountToFineAmount(amountSecondDeposit, assetDecimals(sourceAsset)))
+      ) {
+        throw new Error(
+          'Swap input amount is greater than the second deposit ' + inputAmount.toString(),
+        );
+      }
+      return false;
+    },
+    abortable: true,
+  });
```
Comment (kylezs) [inline bouncer/tests/delta_base_ingress.ts:128]: This is almost identical to the above call, we can make a local function to dedupe

File: bouncer/tests/delta_base_ingress.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  amountToFineAmount,
+  assetDecimals,
+  executeWithTimeout,
+  observeFetch,
+  sleep,
+} from '../shared/utils';
+import { observeEvent } from '../shared/utils/substrate';
+import { killEngines, startEngines } from '../shared/upgrade_network';
+
+let swapsWitnessed: number = 0;
```
Comment (kylezs) [inline bouncer/tests/delta_base_ingress.ts:33]: move inside the function?

**Review by kylezs — COMMENTED** [review]
Nice one 🙌  will be very handy soon, just some cleanliness comments

File: bouncer/tests/delta_base_ingress.ts (hunk 1 above)
Comment (albert-llimos) [reply bouncer/tests/delta_base_ingress.ts]: It was to do it to two different destination chains but that will be tested in other tests so I'll do the change.

File: bouncer/tests/delta_base_ingress.ts (hunk 3 above)
Comment (albert-llimos) [reply bouncer/tests/delta_base_ingress.ts:157]: it's not the same check. Inside the `SwapScheduled` we check that there are no more than two saps and here we check that there not less than two. We are making sure that one swap was seen (aka final number of swaps == 2)

File: bouncer/tests/delta_base_ingress.ts (hunk 5 above)
Comment (albert-llimos) [reply bouncer/tests/delta_base_ingress.ts]: already done in the commit pushed almost at the same time  as the comment

**Review by kylezs — APPROVED** [review]

**Pull request merged**